package.json

```diff
@@ -41,6 +42,7 @@
 	},
 	"homepage": "https://github.com/11ty/eleventy-fetch#readme",
 	"devDependencies": {
+		"@types/node": "^20.14.6",
```

Comment:
This needs to be a true dependency, not just for development. Otherwise when you add eleventy-fetch to your normal dependencies, your package manager is unlikely to download @types/node and the type declaration file will become incomplete.

Reply:
I don't like the idea of declaring this as a full dependency, as that would require all projects, regardless of type support, to pull a rather large type definition. Some bundlers will bundle all dependencies.

While, yes, native TS projects may include this as a dependency, and TS recommends shipping the types in the final bundle, this isn't a native TS project, and most consumers won't be typescript based.

Reply:
This sums up my experiences: https://github.com/Microsoft/types-publisher/issues/81#issuecomment-459314973

Reply:
That’s an interesting sumup of the issue! But it does not seem to land at any specific conclusion 🤔

I am confused how the eleventy-fetch.d.ts is supposed to work if it depends on a type (from @types/node) that is not in my local environment. Surely Typescript is just going to tell me that the following code returns with a type `any` then?

```js
eleventyFetch("", { type: "buffer" });
```

I do understand the issue of incompatible node types, conflicts with the parent, etc. I am now thinking this should maybe be a `peerDependency`, potentially set as `"@types/node": "*"`. That way at least the package manager can inform the end-user that there is an unmet requirement when no node types have been loaded.

I think `*` is a perfectly valid version here, as the code is happy to accept any version of Node.js that supports `Buffer`.

Reply:
I think having an error is better than telling typeless users that they are missing a peer dependency or pulling in a large dependency. Again, focusing on the primary user being a user of 11ty - which lacks types (types were accidently removed recently).

The whole, "don't let perfection prevent progress" idea. There aren't any perfect solutions here, so I'm optimizing for not "harming" existing users, while making TS users slightly better.

Another thing to consider, if a TS user needs the buffer api, wouldn't they already have @types/node installed in the consuming project?